Add qualified identifier changes and new reserved keywords to the release note (#1287)
ballerina-platform/ballerina-release: release-notes/swan-lake-beta4-release-note.md

float f14 = 0x1A.0p4;
```

- Intervening white spaces have been disallowed in the qualified identifier to avoid a parsing ambiguity between the ternary conditional expression and qualified identifier.
  
```ballerina
import ballerina/io;

public function main() {
    io:print("Ballerina"); // Valid.
    io : print("Ballerina"); // compilation error: intervening whitespaces are not allowed in a qualified identifier.
}
```

> With this, `x ? a : b:c` will be parsed now as `x ? a : (b:c)` since the colon with spaces is interpreted only as part of a conditional expression.

Reviewer: ```suggestion
With this, `x ? a : b:c` will now be parsed as `x ? a : (b:c)` since the colon with spaces is interpreted only as part of a conditional expression.
```

float f14 = 0x1A.0p4;
```

> - Intervening white spaces are disallowed in the qualified identifier to avoid parsing ambiguity between ternary conditional expression and qualified identifier.

Reviewer 2: ```suggestion
- Disallowed intervening white spaces in the qualified identifier to avoid parsing an ambiguity between the ternary conditional expression and qualified identifier.
```

  
```ballerina
import ballerina/io;

public function main() {
io:print("Ballerina"); // valid

io : print("Ballerina"); // compilation error: intervening whitespaces are not allowed in qualified identifier
}
```

> With this `x ? a : b:c` will be parsed as `x ? a : (b:c)` since colon with spaces is interpreted only as part of a conditional expression.

Reviewer 2: ```suggestion
With this, `x ? a : b:c` will be parsed now as `x ? a : (b:c)` since the colon with spaces is interpreted only as part of a conditional expression.
```

float f14 = 0x1A.0p4;
```

> - Disallowed intervening white spaces in the qualified identifier to avoid parsing an ambiguity between the ternary conditional expression and qualified identifier.

Reviewer: Shouldn't this be
```suggestion
- Disallowed intervening white spaces in the qualified identifier to avoid a parsing ambiguity between the ternary conditional expression and qualified identifier.
```

Not sure what "to avoid parsing an ambiguity" means.

Reviewer: Also to maintain consistency, I think this should be

```suggestion
- Intervening white spaces have been disallowed in the qualified identifier to avoid a parsing ambiguity between the ternary conditional expression and qualified identifier.
```

  
```ballerina
import ballerina/io;

public function main() {

> io:print("Ballerina"); // valid
> 
> io : print("Ballerina"); // compilation error: intervening whitespaces are not allowed in qualified identifier

Reviewer: ```suggestion
    io:print("Ballerina"); // Valid.

    io : print("Ballerina"); // compilation error: intervening whitespaces are not allowed in a qualified identifier.
```

}
```

With this, `x ? a : b:c` will be parsed now as `x ? a : (b:c)` since the colon with spaces is interpreted only as part of a conditional expression.

- Reserve set of identifiers as ballerina keywords

Identifiers `where`, `join`, `order`, `by`, `equals`, `ascending`, `descending`, `limit`, `outer`, and `select` are now considered as the ballerina reserved keywords. Further, the keyword `join` is allowed to be used in the method call expression as the method name.

```ballerina
// Previously allowed, now disallowed. As an alternative quoted identifier (`'limit`) can be used.
int limit = 5;
// Previously allowed, now disallowed. As an alternative quoted identifier (`string:'join()`) can be used.
string b = string:join(" ", "hello", "world!");
// Continue to allow `join` in method call expression
myObject.join(obj); // allowed

> ``` 

Reviewer: Since we didn't specifically mention allowing these, I don't think we need to specify disallowing these? WDYT?

Author: We have mentioned that in Beta3 release note.

Reviewer: Can you point to it please? Didn't we remove it as discussed in https://github.com/ballerina-platform/ballerina-dev-website/pull/2897#discussion_r725014020?

Author: Removed the section.
